src/core/tools/readFileTool.ts

```diff
@@ -52,69 +47,43 @@ export async function readFileTool(
 			const { maxReadFileLine = 500 } = (await cline.providerRef.deref()?.getState()) ?? {}
 			const isFullRead = maxReadFileLine === -1
 
-			// Check if we're doing a line range read
-			let isRangeRead = false
-			let startLine: number | undefined = undefined
-			let endLine: number | undefined = undefined
-
-			// Check if we have either range parameter and we're not doing a full read
-			if (!isFullRead && (startLineStr || endLineStr)) {
-				isRangeRead = true
-			}
-
-			// Parse start_line if provided
+			// Parse start_line if provided (keep as 1-based)
+			let requestedStartLine: number | undefined = undefined
 			if (startLineStr) {
-				startLine = parseInt(startLineStr)
-
-				if (isNaN(startLine)) {
-					// Invalid start_line
+				requestedStartLine = parseInt(startLineStr)
+				if (isNaN(requestedStartLine)) {
 					cline.consecutiveMistakeCount++
 					cline.recordToolError("read_file")
 					await cline.say("error", `Failed to parse start_line: ${startLineStr}`)
 					pushToolResult(`<file><path>${relPath}</path><error>Invalid start_line value</error></file>`)
 					return
 				}
-
-				startLine -= 1 // Convert to 0-based index
 			}
 
-			// Parse end_line if provided
+			// Parse end_line if provided (keep as 1-based)
+			let requestedEndLine: number | undefined = undefined
 			if (endLineStr) {
-				endLine = parseInt(endLineStr)
-
-				if (isNaN(endLine)) {
-					// Invalid end_line
+				requestedEndLine = parseInt(endLineStr)
+				if (isNaN(requestedEndLine)) {
 					cline.consecutiveMistakeCount++
 					cline.recordToolError("read_file")
 					await cline.say("error", `Failed to parse end_line: ${endLineStr}`)
 					pushToolResult(`<file><path>${relPath}</path><error>Invalid end_line value</error></file>`)
 					return
 				}
-
-				// Convert to 0-based index
-				endLine -= 1
-			}
-
-			const accessAllowed = cline.rooIgnoreController?.validateAccess(relPath)
-
-			if (!accessAllowed) {
```

Comment:
where did access allowed go?

if it moved somewhere centralized, then great but I want to make sure we are not creating regressions